pkg/morph/client/nns.go

```diff
@@ -90,94 +86,64 @@ func (c *Client) NNSHash() (util.Uint160, error) {
 	return *nnsHash, nil
 }
 
-func nnsResolveItems(c *rpcclient.WSClient, nnsHash util.Uint160, domain string) ([]string, error) {
-	var nnsReader = nns.NewReader(invoker.New(c, nil), nnsHash)
-
-	found, err := exists(nnsReader, domain)
+func autoSidechainScope(ws *rpcclient.WSClient, conf *cfg) error {
+	nnsHash, err := nns.InferHash(ws)
 	if err != nil {
-		return nil, fmt.Errorf("could not check presence in NNS contract for %s: %w", domain, err)
+		return fmt.Errorf("InferHash: %w", err)
 	}
 
-	if !found {
-		return nil, ErrNNSRecordNotFound
-	}
+	var nnsReader = nns.NewReader(invoker.New(ws, nil), nnsHash)
 
-	result, err := nnsReader.Resolve(domain, nns.TXT)
+	balanceHash, err := nnsReader.ResolveFSContract(nns.NameBalance)
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("resolving balance: %w", err)
 	}
-	if len(result) == 0 {
-		return nil, errEmptyResultStack
-	}
-	return result, nil
-}
-
-// exists checks domain presence in the NNS contract with given address by
-// calling `ownerOf` method. Returns true if call succeeded only.
-func exists(nnsReader *nns.ContractReader, domain string) (bool, error) {
-	_, err := nnsReader.OwnerOf([]byte(domain))
-
+	cntHash, err := nnsReader.ResolveFSContract(nns.NameContainer)
 	if err != nil {
-		if strings.Contains(err.Error(), "invocation failed") {
-			err = nil
-		}
-		return false, err
+		return fmt.Errorf("resolving container: %w", err)
 	}
-
-	return true, nil
-}
-
-// SetGroupSignerScope makes the default signer scope include all NeoFS contracts.
-// Should be called for side-chain client only.
-func (c *Client) SetGroupSignerScope() error {
-	c.switchLock.RLock()
-	defer c.switchLock.RUnlock()
-
-	if c.inactive {
-		return ErrConnectionLost
-	}
-
-	pub, err := c.contractGroupKey()
+	netmapHash, err := nnsReader.ResolveFSContract(nns.NameNetmap)
 	if err != nil {
-		return err
+		return fmt.Errorf("resolving netmap: %w", err)
 	}
-
-	// Don't change c before everything is OK.
-	cfg := c.cfg
-	cfg.signer = &transaction.Signer{
-		Scopes:        transaction.CustomGroups | transaction.CalledByEntry,
-		AllowedGroups: []*keys.PublicKey{pub},
-	}
-	rpcActor, err := newActor(c.client, c.acc, cfg)
+	neofsIDHash, err := nnsReader.ResolveFSContract(nns.NameNeoFSID)
 	if err != nil {
-		return err
+		return fmt.Errorf("resolving neofsid: %w", err)
+	}
+
+	conf.signer = &transaction.Signer{
+		Scopes: transaction.CalledByEntry | transaction.Rules,
+		Rules: []transaction.WitnessRule{{
+			Action: transaction.WitnessAllow,
+			Condition: &transaction.ConditionAnd{
+				(*transaction.ConditionCalledByContract)(&cntHash),
+				(*transaction.ConditionScriptHash)(&balanceHash),
```

Comment:
what error/log/side effect will be seen when a new dev comes and changes a contract and the morph client breaks?

Reply:
Some invocation will fail, likely even before it's created (test invocation), therefore it'll be seen pretty quickly.